### ISSUE TITLE
Device overheating issues - Video Calls

### ISSUE THREAD

**Issue by bhavinb98** [AgoraIO-Extensions/Agora-Flutter-SDK#758]
Device overheats during video calls. Problem noted in both Android and iPhone devices.

The video quality is already set to low :- Resolution: 320 × 180; frame rate: 15 fps; bitrate: 140 Kbps
No other video enhancements are being used.

Agora version being used :- agora_rtc_engine: ^5.3.0

Edit - Maximum 4 people in video calls.

**Comment by vakrolme** [comment]
Noticed this as well.

**Comment by littleGnAl** [comment]
Which device you are using? media encoding/decoding naturally costs more than normal mobile activities. The impact of low-end mobile phones will be more obvious.

**Comment by bhavinb98** [comment]
iPhone 8, One plus 6, One plus Nord are the devices on which the problem was noted without doubt. Will include more devices if applicable.

Also, only 4 people are in a video call at a time. I believe these devices should have no problem processing video calls with such small number of participants.

**Comment by vakrolme** [comment]
For me it's Samsung Galaxy S10E (Exynos 9820 CPU) which I use for debugging via USB tethering. 

I also use iPhone XS, and I never see it heat anywhere as much, even though the 2 devices have comparable CPUs from the same generation, which might indicate the problem may be Android-specific.

I mostly work with 6.0.0, and at one point while running a test call on 6.0.0-beta.1, I got a notification in the Android tray saying something like **"charging stopped because your device is too hot"**. 

I didn't get that specific message since then, and later versions of 6.0.0 seem to do better, but there are other indications that the 6.0.0 isn't as efficient as 5.x.x yet:

1. CPU usage with 6.0.0 is consistently 25—29%, whereas with 5.x.x it is mostly stays within 15—22% range, on the same codebase and same device.
2. I get 30% less FPS on the same device on 6.0.0 than on 5.3.0 (I already opened a ticket **CUS-26828** about it).
